### DISCUSSION TITLE
Game randomly starts lagging intensively during score conversion

### DISCUSSION THREAD

**Discussion by dekrain** [ppy/osu#26130] [Q&A]
### Type

Performance

### Bug description

After running the latest update, when converting scores to the new format, about 1000 scores were processed fine in the background, but after some point, the game started lagging very hard, even dropping to 0.5 FPS at times. Only the game Update (and consequently, Draw) thread was affected, system didn't even report a CPU spike for the game. The CPU load stopped for a brief moment a couple of times. After all scores finished processing, the performance went back to normal.
Clearing the caches (and thus invoking GC) didn't change anything.

In the meantime I also discovered that the framework performance profiler seems to only account for `Drawable`s, so background tasks were attributed to random bogus UI elements instead.

### Screenshots or videos

One of the moments while this was happening:
![lazer-score-hang](https://github.com/ppy/osu/assets/19158291/752e3635-a091-474f-a3f8-281ed178b901)

### Version

2023.1224.0 (Linux x86-64)

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13767593/compressed-logs.zip)

**Comment by peppy** [comment]
How long did it last, and during what timestamps?

**Reply by dekrain** [reply]
It started about 5-10 minutes since the conversion started; as I said about 1000 scores in. It lasted until all scores finished processing.
